reframe/core/environments.py

```diff
@@ -79,8 +79,9 @@ def load(self):
             if rt.modules_system.is_module_loaded(m):
                 self._preloaded.add(m)
 
-            self._conflicted += rt.modules_system.load_module(m, force=True)
-            for conflict in self._conflicted:
+            module_conflicts = rt.modules_system.load_module(m, force=True)
+            self._conflicted += module_conflicts
+            for conflict in module_conflicts:
```

Comment:
The loop variable should be simply `m`.

Reply:
its a nested loop and the outer one already uses `m`